master/internal/rm/multirm/multirm.go

```diff
@@ -234,12 +234,33 @@ func (m *MultiRMRouter) GetJobQ(rpName rm.ResourcePoolName) (map[model.JobID]*sp
 func (m *MultiRMRouter) GetJobQueueStatsRequest(req *apiv1.GetJobQueueStatsRequest) (
 	*apiv1.GetJobQueueStatsResponse, error,
 ) {
-	resolvedRMName, err := m.getRM(rm.ResourcePoolName(req.ResourcePools[0]))
+	filteredRMs := map[string]rm.ResourceManager{}
+	for _, rName := range req.ResourcePools {
```

Comment:
blocking: when ```req.ResourcePools``` is the empty list or nil then I think the behavior is to get every resource pool

https://github.com/determined-ai/determined/blob/37abc6ce58530a598a1987fc25686acec1247b1e/master/internal/rm/agentrm/agent_resource_manager.go#L303

Is there anywhere else we could have made this same mistake?

Reply:
The rest of the interface is straightforward, but I do want some confirmation on this:
```
// IsReattachableOnlyAfterStarted routes a IsReattachableOnlyAfterStarted call to a specified resource manager/pool.
func (m *MultiRMRouter) IsReattachableOnlyAfterStarted() bool {
	resolvedRMName, err := m.getRM("")
	if err != nil {
		m.syslog.WithError(err)
		return false // Not sure what else to return here.
	}

	return m.rms[resolvedRMName].IsReattachableOnlyAfterStarted()
}
```

Reply:
I think that is fine

another option could be to make a function ```getDefaultRM()``` which just returns the defaultRM and no error so you don't have to think about cases like this